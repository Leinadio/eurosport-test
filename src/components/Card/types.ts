import { Country } from '../../services/graphql/headToHead';

export interface PropsType {
  url: string;
  firstname: string;
  lastname: string;
  shortname: string;
  sex: string;
  age: number;
  height: number;
  weight: number;
  rank: number;
  points: number;
  country: Country;
  last: boolean[];
}