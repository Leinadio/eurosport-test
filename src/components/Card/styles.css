.card {
    padding: 24px 0;
    background: #e7e7e7;
    margin-right: 16px;
    margin-left: 16px;
    margin-bottom: 32px;
}


.card-header {
    display: flex;
    flex-direction: column;
}

@media all and (min-width: 768px) {
    .card-header {
        display: flex;
        flex-direction: column;
    }
}


.card-left {
    margin-right: 16px;
    display: flex;
    flex-direction: column;
}

.card-right {
    margin-top: 24px;
}

.card-img {
    width: 200px;
    margin: auto;

}

.card-content {
    margin-top: 24px;
}

.card-annexe {
    margin-top: 24px;
}

.card-title {
    text-align: center;
    font-size: 24px;
    font-family: Arial;
    color: black;
    margin-bottom: 8px;
}

.card-text {
    text-align: center;
    font-size: 16px;
    font-family: Arial;
    color: black;
    margin-bottom: 8px;
}

.card-span {
    font-weight: bold;
    font-family: Arial;
}

.last {
    display: flex;
    justify-content: space-around;
    flex-wrap: wrap;
    width: 140px;
    margin: auto;
}

.dot-lose {
    width: 20px;
    height: 20px;
    display: block;
    border-radius: 50%;
    background: red;
}

.dot-win {
    width: 20px;
    height: 20px;
    display: block;
    border-radius: 50%;
    background: green;
}

.totalWinsLoses {
    width: 210px;
    margin: auto;
    display: flex;
    justify-content: space-around;
}

.totalWinsLosesValue {
    display: block;
    text-align: center;
}