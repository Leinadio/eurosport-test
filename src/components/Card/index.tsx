import React, { useEffect, useState } from 'react';
import { PropsType } from './types';
import './styles.css'
import { generateRandomNumber } from '../../utils/generateRandomNumber';

function showWinsLoses(data: any) {
  return data.map((el: any) => {
    if (!el) {
      return (
        <span key={`${generateRandomNumber()}`} className="dot-lose">{null}</span>
      )
    }
    return (
      <span key={`${generateRandomNumber()}`} className="dot-win">{null}</span>
    )
  })
}

export default function Card (props: PropsType): JSX.Element {
  const {
    url,
    firstname,
    lastname,
    shortname,
    sex,
    age,
    height,
    weight,
    rank,
    points,
    last,
    country
  } = props;

  const [totalWins, settotalWins] = useState(0);
  const [totalLoses, settotalLoses] = useState(0);
  useEffect(() => {
    const totalWinsCounted: number = last.filter((el: boolean): boolean => el).length
    const totalLosesCounted: number = last.filter((el: boolean): boolean => !el).length
    settotalWins(totalWinsCounted)
    settotalLoses(totalLosesCounted)
  }, [last]);

  return (
    <div className="card">
      <div className="card-header">
        <div className="card-left">
          <img
            src={url}
            alt={`${firstname} ${lastname}`}
            className="card-img"
          />
        </div>
        <div className="card-right">
          <p className="card-text"><span className="card-span">Prenom:</span> {firstname}</p>
          <p className="card-text"><span className="card-span">Nom:</span> {lastname}</p>
          <p className="card-text"><span className="card-span">Shortname:</span> {shortname}</p>
          <p className="card-text"><span className="card-span">Age:</span> {age}</p>
          <p className="card-text"><span className="card-span">Sexe:</span> {sex}</p>
          <p className="card-text"><span className="card-span">Country:</span> {country.code}</p>
          <img
            src={country.picture.url}
            alt={`${country.code}`}
            className="card-country-img"
          />
        </div>
      </div>
      <div className="card-content">
        <h1 className="card-title">Statistiques</h1>
        <p className="card-text"><span className="card-span">Height:</span> {height}m</p>
        <p className="card-text"><span className="card-span">Weight:</span> {weight*0.001}Kg</p>
        <p className="card-text"><span className="card-span">Points:</span> {points}</p>
        <p className="card-text"><span className="card-span">Rank:</span> {rank}</p>
      </div>
      <div className="card-annexe">
        <h1 className="card-title">Wins/Loses</h1>
        <div className="totalWinsLoses">
          <p className="card-text">Total Wins: <span className="card-span totalWinsLosesValue">{totalWins}</span></p>
          <p className="card-text">Total Loses: <span className="card-span totalWinsLosesValue">{totalLoses}</span></p>
        </div>
        <div className="last">
          {showWinsLoses(last)}
        </div>
      </div>
    </div>

  )
}