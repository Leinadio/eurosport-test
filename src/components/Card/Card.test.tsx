import React from 'react';
import renderer from 'react-test-renderer';
import Card from './'

const props = {
  url: 'test',
  firstname: 'jon',
  lastname: 'snow',
  shortname: 'js',
  sex: 'man',
  age: 25,
  height: 185,
  weight: 8500,
  points: 3200,
  rank: 1,
  last: [true, true, true],
  country: {
    code: 'SUI',
    picture: {
      url: 'imgflag'
    }
  }
}
describe('showWinsLoses', function () {
  it('should display red dot', () => {
    const tree = renderer
      .create(<Card {...props} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});