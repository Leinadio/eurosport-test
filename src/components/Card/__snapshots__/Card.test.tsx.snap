// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`showWinsLoses should display red dot 1`] = `
<div
  className="card"
>
  <div
    className="card-header"
  >
    <div
      className="card-left"
    >
      <img
        alt="jon snow"
        className="card-img"
        src="test"
      />
    </div>
    <div
      className="card-right"
    >
      <p
        className="card-text"
      >
        <span
          className="card-span"
        >
          Prenom:
        </span>
         
        jon
      </p>
      <p
        className="card-text"
      >
        <span
          className="card-span"
        >
          Nom:
        </span>
         
        snow
      </p>
      <p
        className="card-text"
      >
        <span
          className="card-span"
        >
          Shortname:
        </span>
         
        js
      </p>
      <p
        className="card-text"
      >
        <span
          className="card-span"
        >
          Age:
        </span>
         
        25
      </p>
      <p
        className="card-text"
      >
        <span
          className="card-span"
        >
          Sexe:
        </span>
         
        man
      </p>
      <p
        className="card-text"
      >
        <span
          className="card-span"
        >
          Country:
        </span>
         
        SUI
      </p>
      <img
        alt="SUI"
        className="card-country-img"
        src="imgflag"
      />
    </div>
  </div>
  <div
    className="card-content"
  >
    <h1
      className="card-title"
    >
      Statistiques
    </h1>
    <p
      className="card-text"
    >
      <span
        className="card-span"
      >
        Height:
      </span>
       
      185
      m
    </p>
    <p
      className="card-text"
    >
      <span
        className="card-span"
      >
        Weight:
      </span>
       
      8.5
      Kg
    </p>
    <p
      className="card-text"
    >
      <span
        className="card-span"
      >
        Points:
      </span>
       
      3200
    </p>
    <p
      className="card-text"
    >
      <span
        className="card-span"
      >
        Rank:
      </span>
       
      1
    </p>
  </div>
  <div
    className="card-annexe"
  >
    <h1
      className="card-title"
    >
      Wins/Loses
    </h1>
    <div
      className="totalWinsLoses"
    >
      <p
        className="card-text"
      >
        Total Wins: 
        <span
          className="card-span totalWinsLosesValue"
        >
          0
        </span>
      </p>
      <p
        className="card-text"
      >
        Total Loses: 
        <span
          className="card-span totalWinsLosesValue"
        >
          0
        </span>
      </p>
    </div>
    <div
      className="last"
    >
      <span
        className="dot-win"
      />
      <span
        className="dot-win"
      />
      <span
        className="dot-win"
      />
    </div>
  </div>
</div>
`;
